components/script/dom/webgl/webglframebuffer.rs

```diff
@@ -776,7 +776,9 @@ impl WebGLFramebuffer {
                 _ => return Err(WebGLError::InvalidOperation),
             }
 
-            let context = self.upcast::<WebGLObject>().context();
+            let Some(context) = self.upcast::<WebGLObject>().context() else {
```

Comment:
Seem like in many places we can remove the `::<WebGLObject>` annotation as `WebGLObject` is the only ancestor interface and the type checker is able to infer that.